skbio/stats/composition.py

```diff
@@ -896,9 +961,27 @@ def ancom(table, grouping,
         else:
             nu = cutoff[4]
         reject = (W >= nu*n_feat)
-    labs = mat.columns
-    return pd.DataFrame({'W': pd.Series(W, index=labs),
-                         'reject': pd.Series(reject, index=labs)})
+
+    feat_ids = mat.columns
+    ancom_df = pd.DataFrame(
+        {'W': pd.Series(W, index=feat_ids),
+         'Reject null hypothesis': pd.Series(reject, index=feat_ids)})
+
+    if len(percentiles) == 0:
+        return ancom_df, pd.DataFrame()
+    else:
+        data = []
+        columns = []
+        for group in groups:
+            feat_dists = mat[grouping == group]
```

Comment:
I was playing around with tests locally and wasn't receiving the `UserWarning` as I expected (we have a test case that should hit it). The reason is that `table` is aligned to whatever order `grouping` is in, so `cats` is just a copy of `grouping`. Thus, using `grouping` vs. `cats` here is equivalent and we don't hit the pandas codepath I suspected.

`DataFrame.align` doesn't document how the index alignment will be performed. It appears to align the data frame (i.e. the object the `align` method is called on) to the input series, returning the aligned data frame and a copy of the input series. Since this is undocumented behavior and might change across pandas versions or in the future, it's safer to use `cats` here, even though `cats` and `grouping` are currently equivalent.